### PR TITLE
"Curve Mapper": add Curve output.

### PR DESCRIPTION
**Pull request by portnov**
This adds a "Curve" output to the Curve Mapper node.

Screenshot_20200428_224813

## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

### CODE REVIEW

File: nodes/number/curve_mapper.py [hunk 1]
```diff
@@ -132,15 +132,25 @@ def sv_copy(self, node):
...
-        n_id = self.node_id
-        evaluate = get_evaluator(node_group_name, 'RGB Curves'+n_id)
+        curve_node_name = self._get_curve_node_name()
+        evaluate = get_evaluator(node_group_name, curve_node_name)
+        curve_node = get_valid_node(node_group_name, curve_node_name, CURVE_NODE_TYPE)
+
+        curve = SvScalarFunctionCurve(evaluate)
+        curve.u_bounds = (curve_node.mapping.clip_min_x, curve_node.mapping.clip_max_x)
+        if 'Curve' in self.outputs:
+            self.outputs['Curve'].sv_set([curve])
```
Comment (vicdoval) [inline nodes/number/curve_mapper.py:152]: `if 'Curve' in self.outputs and self.outputs['Curve'].is_linked` could be before the curve is created

**Comment by vicdoval** [comment]
You were fast :)
I was coding this #3137, maybe you can add the new functions in this PR and i cancel the other.

btw thanks for the cleaning.
Everything seems nice just the little comment I did on the code, maybe is a fast operation but why doing it if is not demanded....

File: nodes/number/curve_mapper.py (hunk 1 above)
Comment (portnov) [reply nodes/number/curve_mapper.py:152]: Ok, I agree. These lines are really fast, but the code will be cleaner.

**Comment by portnov** [comment]
I think different features better be in different PRs.

**Pull request merged**